basics.js

```diff
@@ -1,24 +1,85 @@
 /* Create a `myName` variable and assign it a String value */
 
+function Person (name) {
+
+    var myName = name;
+   return myName;
+ }
+  console.log(Person('Gail'));
+
+
 /* Create a `person` variable and give it 2 properties,
  * `name`, assign it the same name as before,
  * as well as an `age` (number);
  */
+var Person = {
+   name: 'Gail',
+   age: 21,
+};
+
 
 /* Create a function called `greet`,
  * it should take a 1 parameter, `name`
  * and it should print "Hello, my name is {name}"
  */
+function greet (name) {
+return "Hello, my name is " + name;
 
+}
+console.log(greet('Joe'));
 /* Create a variable called `canDrive`,
  * if it should be true if your person object is at least 16 years old
  */
+function canDrive (age) {
 
+if (age >= 16) {return true;}
+else if (age < 16) {return false;}
+}
+
+console.log(canDrive(29));
 /* Create an array called `dataTypes` with atleast 1 of every data type;
  * (there are 6 different data types);
  */
 
+function findDataType (dataTypes) {
```

Comment:
Nice attempt. This portion of the test is asking you to just create an array called `dataTypes` and to add data to it.
No need to create a function, or a for loop.

``javascript
var dataTypes = [];  // Put values into this array
```